src/utils.rs

```diff
@@ -119,4 +137,59 @@ mod test {
             assert!(x.iter().all(|xi| *xi == 0));
         }
     }
+
+    #[test]
+    fn test_add_le_zero() {
+        for i in 1usize..256 {
+            let mut x = vec!(0u8; i);
+            let mut y = vec!(0u8; i);
```

Comment:
Doesn't need to be mutable

Reply:
It does because of the next line. Or how else I can initialize a vector from all but one zeros?